pkg/xds/context/mesh_context_builder.go

```diff
@@ -482,10 +468,13 @@ func (m *meshContextBuilder) resolveTLSReadiness(
 	}
 }
 
-func (m *meshContextBuilder) decorateWithCrossMeshResources(ctx context.Context, resources Resources) error {
+func (m *meshContextBuilder) decorateWithCrossMeshResources(ctx context.Context, meshName string, resources Resources) error {
 	// Expand with crossMesh info
 	otherMeshesByName := map[string]*core_mesh.MeshResource{}
-	for _, m := range resources.OtherMeshes().GetItems() {
+	for _, m := range resources.Meshes().GetItems() {
+		if m.GetMeta().GetName() == meshName {
```

Comment:
Comment to explain we're skipping self

Reply:
Changed this code to use `OtherMeshes(localMesh)`, probably the comment now is less needed.